bc_obps/reporting/tests/api/test_report_supplementary_version_api.py

```diff
@@ -1,74 +1,95 @@
+        ],
+    )
     @patch(
-        "reporting.service.report_supplementary_version_service.ReportSupplementaryVersionService.create_report_supplementary_version"
+        "reporting.service.report_supplementary_version_service.ReportSupplementaryVersionService"
+        ".create_report_supplementary_version"
     )
-    def test_returns_data_as_provided_by_create_report_supplementary_version(
-        self, mock_create_report_supplementary_version: MagicMock | AsyncMock
+    @patch("service.report_version_service.ReportVersionService.create_report_version")
+    def test_create_version_based_on_purpose(
+        self,
+        mock_create_report_version: MagicMock,
+        mock_create_supplementary: MagicMock,
+        existing_purpose,
+        new_purpose,
+        expected_called_method,
     ):
-        # Arrange: Mock service method return value
-        mock_create_report_supplementary_version.return_value = self.new_report_version
+        report_version = self.old_report_version
+
+        report_version.report.operation.registration_purpose = existing_purpose
+        report_version.report.operation.save()
+
+        baker.make_recipe(
+            "reporting.tests.utils.report_operation",
+            report_version=report_version,
+            registration_purpose=new_purpose,
+        )
+
+        if expected_called_method == "supplementary":
+            mock_create_supplementary.return_value = self.new_report_version
+        else:
+            mock_create_report_version.return_value = self.new_report_version
 
-        # Act: Authorize user and perform POST request
-        endpoint_under_test = "create_report_supplementary_version"
-        endpoint_under_test_kwargs = {"version_id": self.old_report_version.id}
         response = TestUtils.mock_post_with_auth_role(
             self,
             "industry_user",
             self.content_type,
-            {},  # No request payload expected
+            {},
             custom_reverse_lazy(
-                endpoint_under_test,
-                kwargs=endpoint_under_test_kwargs,
+                "create_report_supplementary_version",
+                kwargs={"version_id": report_version.id},
             ),
         )
 
-        # Assert: Check the response status and content
         assert response.status_code == 201
         assert response.json() == self.new_report_version.id
 
-        # Assert: Ensure the service method was called with the correct version_id
-        mock_create_report_supplementary_version.assert_called_once_with(self.old_report_version.id)
+        if expected_called_method == "supplementary":
+            mock_create_supplementary.assert_called_once_with(report_version.id)
+            mock_create_report_version.assert_not_called()
+        else:
+            mock_create_report_version.assert_called_once_with(report_version.report)
+            mock_create_supplementary.assert_not_called()
```

Comment:
Could we get an extra test that uses "real" (model bakery) database instances for report versions, and verify that the new version is created with or without the emission/production records?
It would be a small integration test, and make sure that the behaviour we want happens all the way down